src/System.Management.Automation/engine/PSVersionInfo.cs

```diff
@@ -469,16 +475,19 @@ public SemanticVersion(int major, int minor, int patch, string preReleaseLabel,
         {
             if (!string.IsNullOrEmpty(preReleaseLabel))
             {
-                if (!Regex.IsMatch(preReleaseLabel, LabelUnitRegEx)) throw new FormatException(nameof(preReleaseLabel));
```

Comment:
It looks like the old API would throw if the labels don't meet the semver requirement, but this new change will simply ignore them.  Was it a deliberate decision to not throw?

Reply:
Oops, my comment above was not addressed :-(

Reply:
Which requirement?  The tests to verify that we throw still pas.

Reply:
Can you add test cases for the constructor, where the pre-release label is invalid and verify that we throw?

It's odd that we don't have consistent behavior for the build label.